src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py

```diff
@@ -88,34 +88,37 @@ def update_step(
         :param inversion: The type of subspace inversion used in the algorithm, defaults
             to exact.
         """
-        S = sensitivity_matrix
-        A = centered_anomaly_matrix
+        parameters = parameter_ensemble.shape[0]
+        realizations = parameter_ensemble.shape[1]
+        responses = response_ensemble.shape[0]
+        parameter_ensemble = parameter_ensemble
         if step_length is None:
             step_length = self._get_steplength(self._module_data.iteration_nr)
         if noise is None:
-            noise = np.random.rand(*S.shape)
+            noise = np.random.rand(responses, realizations)
         if ensemble_mask is None:
             ensemble_mask = np.array([True] * self._ensemble_size)
         if observation_mask is None:
             observation_mask = np.array([True] * len(observation_values))
 
         E = make_E(observation_errors, noise)
         R = np.identity(len(observation_errors), dtype=np.double)
-        D = make_D(observation_values, E, S)
+        D = make_D(observation_values, E, response_ensemble)
         D = (D.T / observation_errors).T
         E = (E.T / observation_errors).T
-        S = (S.T / observation_errors).T
+        response_ensemble = (response_ensemble.T / observation_errors).T
 
         init_update(self._module_data, ensemble_mask, observation_mask)
 
-        if projection and (A.shape[0] < A.shape[1] - 1):
-            AA_projection = _compute_AA_projection(A)
-            S = S @ AA_projection
+        if projection and (parameters < realizations - 1):
+            AA_projection = _compute_AA_projection(parameter_ensemble)
+            response_ensemble = response_ensemble @ AA_projection
 
         update_A(
             self._module_data,
-            A,
-            S,
+            parameter_ensemble,
+            (response_ensemble - response_ensemble.mean(axis=1, keepdims=True))
```

Comment:
Does this deserve a comment?, same in ensemble_smoother

Reply:
Yeah that got merged as part of the other PR 😢

Reply:
Not sure.
I've added this description to `updateA` and `makeX` in `ies.cpp`:

```c++
/**
 * @param Y Predicted ensemble anomalies normalized by sqrt(N-1),
 *          where N is the number of realizations.
 *          See line 4 of Algorithm 1 and Eq. 30.
 */
```

These function are called from `_iterative_ensemble_smoother` and `_ensemble_smoother` which means that users of the `iterative_ensemble_smoother` package do not have to think about this centering and normalization.
Not sure if it's a good idea to duplicate the information in the docstring.
What do you think?